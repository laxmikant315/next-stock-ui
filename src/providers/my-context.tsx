import React, { useContext } from "react";

const MyContext = React.createContext({name:'Laxmikant',age:12,myVal:"nothing"});
export default MyContext;
export const useMyContext = () => useContext(MyContext);
