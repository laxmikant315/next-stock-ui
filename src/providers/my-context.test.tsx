import React from "react";

import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import MyContext, { useMyContext } from "./my-context";
import * as MyContextModule from "./my-context";

Enzyme.configure({ adapter: new Adapter() });

export const MyComponent = () => {
  const { myVal,name } = useMyContext();

  return <div data-test="my-component">{myVal}{name}</div>;
};

it("renders the correct text", () => {
  jest.spyOn(MyContextModule, "useMyContext").mockImplementation(() => ({
    myVal: "foobar",
    name:'Sandesh'
  }));

  const wrapper = shallow(
    // <MyContext.Provider>
      <MyComponent />
    // </MyContext.Provider>
  )

  expect(wrapper).toMatchSnapshot()
});
