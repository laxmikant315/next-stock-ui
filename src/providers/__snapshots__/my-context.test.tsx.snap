// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct text 1`] = `
<div
  data-test="my-component"
>
  foobar
  Sandesh
</div>
`;
